src/pkg/datastructs/circularbuffer.go

```diff
@@ -0,0 +1,52 @@
+package datastructs
```

Comment:
I would probably just create a specific package for this circularbuffer. 

```suggestion
package circularbuffer
```